### PR TITLE
Near typescript package integration in chain abstraction layer

### PR DESCRIPTION
**Pull request by tima-t**

### CODE REVIEW

File: packages/near-rpc-provider/lib/NearRpcProvider.ts [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    const blockNumber = await this.getBlockHeight(tx.transaction_outcome.block_hash)
+    return { ...tx, blockNumber }
+  }
+
+  async getGasPrice() {
+    const result = await this._rpcQuery('gas_price', [null])
+    return get(result, 'gas_price')
+  }
+
+  async getBalance(addresses: string[]): Promise<BigNumber> {
```
Comment (monokh) [inline packages/near-rpc-provider/lib/NearRpcProvider.ts]: `getBalance` now takes `(addresses: (Address | string)[])`

File: packages/near-rpc-provider/lib/NearRpcProvider.ts [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+    try {
+      if (!isArray(addresses)) {
```
Comment (monokh) [inline packages/near-rpc-provider/lib/NearRpcProvider.ts]: this is not needed. always should be array

File: packages/near-rpc-provider/lib/NearRpcProvider.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+      return new BigNumber(balance.available)
+    } catch (err) {
+      if (err.message && err.message.includes('does not exist while viewing')) {
+        return new BigNumber(0)
+      }
+      throw err
+    }
+  }
+
+  async isAddressUsed(address: string): Promise<boolean> {
```
Comment (monokh) [inline packages/near-rpc-provider/lib/NearRpcProvider.ts]: This method no longer exists on the other providers, we never used it so was removed in the latest version

File: packages/near-swap-find-provider/lib/NearSwapFindProvider.ts [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+      const normalizedTransactions = {} as { [key: string]: near.NearSwapTransaction }
+
+      for (const tx of transactions) {
+        normalizedTransactions[tx.hash] = {
+          ...normalizedTransactions[tx.hash],
+          ...this.normalizeTransactionResponse(tx)
+        }
+      }
+
+      const tx = Object.values(normalizedTransactions).find(predicate) as near.NearSwapTransaction
```
Comment (monokh) [inline packages/near-swap-find-provider/lib/NearSwapFindProvider.ts]: Hmm why does it need to be cast here? It should be that type

File: packages/near-swap-find-provider/lib/NearSwapFindProvider.ts [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+      if (tx) {
+        const currentHeight = await this.getMethod('getBlockHeight')()
+        const txBlockHeight = await this.getMethod('getBlockHeight')(tx.blockHash)
+        tx.confirmations = currentHeight - txBlockHeight
+        return { ...tx, secret: tx.swap.secret, _raw: tx } as Transaction<near.NearSwapTransaction>
```
Comment (monokh) [inline packages/near-swap-find-provider/lib/NearSwapFindProvider.ts]: Might be wrong here, but doesn't casting like this with `as` cause the type checking to be ignored? 
i'm worried some fields will be incorrect by doing it like this

File: packages/near-rpc-provider/lib/NearRpcProvider.ts (hunk 1 above)
Comment (kraikov) [reply packages/near-rpc-provider/lib/NearRpcProvider.ts]: Should be fine now. I've also improved the way balance is calculated.

File: packages/near-rpc-provider/lib/NearRpcProvider.ts (hunk 2 above)
Comment (kraikov) [reply packages/near-rpc-provider/lib/NearRpcProvider.ts]: Thanks. Should be fixed now.

File: packages/near-rpc-provider/lib/NearRpcProvider.ts (hunk 3 above)
Comment (kraikov) [reply packages/near-rpc-provider/lib/NearRpcProvider.ts]: Thanks, wasn't aware of that. It's now removed.

File: packages/near-swap-find-provider/lib/NearSwapFindProvider.ts (hunk 4 above)
Comment (kraikov) [reply packages/near-swap-find-provider/lib/NearSwapFindProvider.ts]: Yeah, that casting was redundant.

File: packages/near-swap-find-provider/lib/NearSwapFindProvider.ts (hunk 5 above)
Comment (kraikov) [reply packages/near-swap-find-provider/lib/NearSwapFindProvider.ts]: Yeah, that casting was redundant.

**Pull request merged**